src/custom/float.rs

```diff
@@ -0,0 +1,152 @@
+
+use super::{CodegenExtension, CodegenExtsMap};
+
+struct FloatTypesCodegenExtension;
+
+impl<'c, H: HugrView> CodegenExtension<'c, H> for FloatTypesCodegenExtension {
+    fn extension(&self) -> hugr::extension::ExtensionId {
+        float_types::EXTENSION_ID
+    }
+
+    fn llvm_type(
+        &self,
+        context: &crate::types::TypingSession<'c, H>,
+        hugr_type: &hugr::types::CustomType,
+    ) -> anyhow::Result<inkwell::types::BasicTypeEnum<'c>> {
+        if hugr_type == &FLOAT64_CUSTOM_TYPE {
+            Ok(context.iw_context().f64_type().as_basic_type_enum())
+        } else {
+            Err(anyhow!(
+                "FloatCodegenExtension: Unsupported type: {}",
+                hugr_type
+            ))
+        }
+    }
+
+    fn emitter<'a>(
+        &self,
+        _context: &'a mut crate::emit::func::EmitFuncContext<'c, H>,
+    ) -> Box<dyn crate::emit::EmitOp<'c, hugr::ops::CustomOp, H> + 'a> {
+        Box::new(NullEmitLlvm)
+    }
+
+    fn supported_consts(&self) -> HashSet<TypeId> {
+        [TypeId::of::<ConstF64>()].into_iter().collect()
+    }
+
+    fn load_constant(
+        &self,
+        context: &mut EmitFuncContext<'c, H>,
+        konst: &dyn hugr::ops::constant::CustomConst,
+    ) -> Result<Option<BasicValueEnum<'c>>> {
+        let Some(k) = konst.downcast_ref::<ConstF64>() else {
+            return Ok(None);
+        };
+        let ty: FloatType<'c> = context
+            .llvm_type(&k.get_type())?
+            .try_into()
+            .map_err(|_| anyhow!("Failed to get type of ConstF64 as FloatType"))?;
+        Ok(Some(ty.const_float(k.value()).as_basic_value_enum()))
+    }
+}
+
+struct FloatOpsCodegenExtension;
+
+impl<'c, H: HugrView> CodegenExtension<'c, H> for FloatOpsCodegenExtension {
+    fn extension(&self) -> hugr::extension::ExtensionId {
+        float_ops::EXTENSION_ID
+    }
+
+    fn llvm_type(
+        &self,
+        _context: &crate::types::TypingSession<'c, H>,
+        hugr_type: &hugr::types::CustomType,
+    ) -> anyhow::Result<inkwell::types::BasicTypeEnum<'c>> {
+        Err(anyhow!(
+            "FloatOpsCodegenExtension: unsupported type: {hugr_type}"
+        ))
+    }
+
+    fn emitter<'a>(
+        &self,
+        context: &'a mut crate::emit::func::EmitFuncContext<'c, H>,
+    ) -> Box<dyn crate::emit::EmitOp<'c, hugr::ops::CustomOp, H> + 'a> {
+        Box::new(FloatOpEmitter(context))
+    }
+}
+
+// we allow dead code for now, but once we implement the emitter, we should
+// remove this
+#[allow(dead_code)]
+struct FloatOpEmitter<'c, 'd, H: HugrView>(&'d mut EmitFuncContext<'c, H>);
+
+impl<'c, H: HugrView> EmitOp<'c, CustomOp, H> for FloatOpEmitter<'c, '_, H> {
+    fn emit(&mut self, args: EmitOpArgs<'c, CustomOp, H>) -> Result<()> {
+        use hugr::ops::NamedOp;
+        let name = args.node().name();
+        #[allow(clippy::match_single_binding)]
+        match name.as_str() {
+            n => Err(anyhow!("FloatOpEmitter: unknown op: {n}")),
```

Comment:
That's a curious way to say `anyhow!("FloatOpEmitter: unknown op: {}", name.as_str())`